chromium/background.js

```diff
@@ -2,45 +2,18 @@
 
 (function(exports) {
 
-/**
- * Load a file packaged with the extension
- *
- * @param url: a relative URL to local file
- */
-function loadExtensionFile(url, returnType) {
-  var xhr = new XMLHttpRequest();
-  // Use blocking XHR to ensure everything is loaded by the time
-  // we return.
-  xhr.open("GET", chrome.extension.getURL(url), false);
-  xhr.send(null);
-  // Get file contents
-  if (xhr.readyState != 4) {
-    return;
-  }
-  if (returnType === 'xml') {
-    return xhr.responseXML;
-  }
-  if (returnType === 'json') {
-    return JSON.parse(xhr.responseText);
-  }
-  return xhr.responseText;
-}
+let all_rules = new rules.RuleSets();
 
+async function initialize(){
+  await store.initialize();
+  await initializeStoredGlobals();
+  all_rules.initialize();
```

Comment:
Just to make sure I understand this:  `await store.init();` will resolve/finish before `await initializeStoredGlbs();` is run?

Reply:
@cowlicks Indeed.

Reply:
Better do this:
```js
store.initialize()
  .then(initializeStoredGlobals)
  .then(() => all_rules.initialize());
```

Reply:
@cowlicks that's right, in ES6 the `await` keyword will wait for the returned promise will resolve before running the next line.  It's a nice way to avoid callback hell in ES6.

To do parallel, you can do:

```js
await Promise.all([asyncFunction1(), asyncFunction2()]);
```